cpp/lib/ClientBindUploader.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing, Inc. All rights reserved.
+ */
+
+
+#include <sstream>
+#include <vector>
+
+#include "ClientBindUploader.hpp"
+#include "../logger/SFLogger.hpp"
+#include "snowflake/basic_types.h"
+#include "snowflake/SF_CRTFunctionSafe.h"
+#include "../util/SnowflakeCommon.hpp"
+#include "snowflake/Exceptions.hpp"
+#include "client_int.h"
+#include "results.h"
+#include "error.h"
+
+namespace Snowflake
+{
+namespace Client
+{
+ClientBindUploader::ClientBindUploader(SF_STMT *sfstmt,
```

Comment:
BindUploader is ported from ODBC and refactored so it can be reused on both libsfclient and ODBC side. When reusing it on ODBC side there would be ODBCBindUploader class to have ODBC specific implementation.